ESSArch_Core/tasks_util.py

```diff
@@ -0,0 +1,260 @@
+    try:
+        arcname = '%s_label.xml' % medium.medium_id
+
+        if medium.format not in [100, 101]:
+            if tape_empty(tape_drive.device):
+                create_tape_label(medium, xmlfile.name)
+                rewind_tape(tape_drive.device)
+                write_to_tape(tape_drive.device, xmlfile.name, arcname=arcname)
+            else:
+                tar = tarfile.open(tape_drive.device, 'r|')
+                first_member = tar.getmembers()[0]
+                tar.close()
+                rewind_tape(tape_drive.device)
+
+                if first_member.name.endswith('_label.xml'):
+                    tar = tarfile.open(tape_drive.device, 'r|')
+                    xmlstring = tar.extractfile(first_member).read()
+                    tar.close()
+                    if not verify_tape_label(medium, xmlstring):
+                        raise ValueError('Tape contains invalid label file')
+                elif first_member.name == 'reuse':
+                    create_tape_label(medium, xmlfile.name)
+                    rewind_tape(tape_drive.device)
+                    write_to_tape(tape_drive.device, xmlfile.name, arcname=arcname)
+                else:
+                    raise ValueError('Tape contains unknown information')
+
+                rewind_tape(tape_drive.device)
+    except BaseException:
+        StorageMedium.objects.filter(pk=medium.pk).update(status=100)
+        TapeDrive.objects.filter(pk=drive_id).update(locked=False, status=100)
+        TapeSlot.objects.filter(slot_id=slot_id).update(status=100)
+        raise
+    finally:
+        xmlfile.close()
+        TapeDrive.objects.filter(pk=drive_id).update(locked=False)
+
+
+def validate_file_format(request_id, filename, format_name, format_registry_key, format_version):
+    """
+    Validates the format of the given file
+    """
+    task = ProcessTask.objects.values(
+        'information_package_id', 'responsible_id'
+    ).get(pk=request_id)
+
+    t = ProcessTask.objects.create(
+        name="ESSArch_Core.tasks.IdentifyFileFormat",
+        params={
+            "filename": filename,
+        },
+        information_package_id=task.get('information_package_id'),
+        responsible_id=task.get('responsible_id'),
+    )
+
+    actual_format_name, actual_format_version, actual_format_registry_key = t.run().get()
+
+    if format_name:
+        assert actual_format_name == format_name, (
+            "format name for %s is not valid, (%s != %s)" % filename, format_name, actual_format_name
+        )
+
+    if format_version:
+        assert actual_format_version == format_version, "format version for %s is not valid" % filename
+
+    if format_registry_key:
+        assert actual_format_registry_key == format_registry_key, (
+            "format registry key for %s is not valid" % filename
+        )
+
+    return "Success"
+
+
+def validate_files(self_ip, responsible, ip, rootdir, validate_fileformat, validate_integrity, xmlfile):
```

Comment:
We only need one information package

```suggestion
def validate_files(ip, responsible, rootdir, validate_fileformat, validate_integrity, xmlfile):
```